assets/js/modules/analytics/components/module/ModuleAcquisitionChannelsWidget/PieChart.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * PieChart component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { DATE_RANGE_OFFSET, STORE_NAME } from '../../../datastore/constants';
+import GoogleChart from '../../../../../components/GoogleChart';
+import { extractAnalyticsDataForPieChart } from '../../../util';
+import { CORE_USER } from '../../../../../googlesitekit/datastore/user/constants';
+import PreviewBlock from '../../../../../components/PreviewBlock';
+
+const { useSelect } = Data;
+
+export default function PieChart() {
+	const {
+		hasFinishedResolution,
+		report,
+		error,
+	} = useSelect( ( select ) => {
+		const dates = select( CORE_USER ).getDateRangeDates( { offsetDays: DATE_RANGE_OFFSET } );
+		const reportArgs = {
+			...dates,
+			dimensions: 'ga:channelGrouping',
+			metrics: [
+				{
+					expression: 'ga:sessions',
+					alias: 'Sessions',
+				},
+				{
+					expression: 'ga:users',
+					alias: 'Users',
+				},
+				{
+					expression: 'ga:newUsers',
+					alias: 'New Users',
+				},
+			],
+			orderby: [
+				{
+					fieldName: 'ga:users',
+					sortOrder: 'DESCENDING',
+				},
+			],
+			limit: 10,
+		};
+
+		return {
+			report: select( STORE_NAME ).getReport( reportArgs ),
+			hasFinishedResolution: select( STORE_NAME ).hasFinishedResolution( 'getReport', [ reportArgs ] ),
+			error: select( STORE_NAME ).getErrorForSelector( 'getReport', [ reportArgs ] ),
+		};
+	} );
+
+	if ( ! report || error || ! hasFinishedResolution ) {
+		return <PreviewBlock width="282px" height="282px" shape="circular" />;
+	}
```

Comment:
We don't need a preview block here. Let's move it to the index.js file before the preview table element to replicate the widget layout.

Reply:
Wouldn't this approach replicate the legacy widget? https://github.com/google/site-kit-wp/blob/f558ad18e6992ba12904a2e584dc7a7140bcff59/assets/js/modules/analytics/components/dashboard/LegacyDashboardAcquisitionPieChart.js#L146

Reply:
@ivankruchkoff we need to create the correct loading state in the index.js file. It should replicate shapes of children components.

Reply:
Updated in https://github.com/google/site-kit-wp/pull/3035/commits/82171c42c4724d042ae2483d7b5a8519a0d4ac08 

Storybook test:
https://google.github.io/site-kit-wp/storybook/pull/3035/?path=/story/analytics-module-components-module-page-acquisition-channels-widget--loading